tests/test_23_04_2025.py

```diff
@@ -0,0 +1,169 @@
+    vebo = get_vebo()
+    ao = get_ao()
+    cs_fee_oracle = get_cs_fee_oracle()
+    csm = get_csm()
+
+    # Before voting tests
+    # 1),3) Aragon agent doesnt have MANAGE_CONSENSUS_VERSION_ROLE on AO
+    check_aragon_doesnt_have_manage_consensus_role_on_oracle(ao)
+    # 2) Accounting Oracle consensus version equals to 2 before voting
+    assert ao.getConsensusVersion() == 2
+    # 4),6) Aragon agent doesnt have MANAGE_CONSENSUS_VERSION_ROLE on Vebo
+    check_aragon_doesnt_have_manage_consensus_role_on_oracle(vebo)
+    # 5) Vebo consensus version equals to 2 before voting
+    assert vebo.getConsensusVersion() == 2
+    # 7),9) Aragon agent doesnt have MANAGE_CONSENSUS_VERSION_ROLE on CS Fee oracle
+    check_aragon_doesnt_have_manage_consensus_role_on_oracle(cs_fee_oracle)
+    # 8) CS fee oracle consensus version equals to 1 before voting
+    assert cs_fee_oracle.getConsensusVersion() == 1
+    # 10) Old CS Verifier has VERIFIER_ROLE role on CSM before voting
+    assert csm.hasRole(VERIFIER_ROLE, CS_VERIFIER_ADDRESS_OLD)
+    # 11) New CS Verifier doesn't have VERIFIER_ROLE role on CSM before voting
+    assert not csm.hasRole(VERIFIER_ROLE, CS_VERIFIER_ADDRESS)
+
+    # START VOTE
+    if len(vote_ids_from_env) > 0:
+        (vote_id,) = vote_ids_from_env
+    else:
+        tx_params = {"from": LDO_HOLDER_ADDRESS_FOR_TESTS}
+        vote_id, _ = start_vote(tx_params, silent=True)
+
+    voting = get_voting()
+
+    vote_tx = helpers.execute_vote(accounts, vote_id, voting)
+
+    print(f"voteId = {vote_id}, gasUsed = {vote_tx.gas_used}")
+
+    # After voting tests
+    # 1),3) Aragon agent doesnt have MANAGE_CONSENSUS_VERSION_ROLE on AO
+    check_aragon_doesnt_have_manage_consensus_role_on_oracle(ao)
+    # 2) Accounting Oracle consensus version equals to 3 after voting
+    assert ao.getConsensusVersion() == 3
+    # 4),6) Aragon agent doesnt have MANAGE_CONSENSUS_VERSION_ROLE on Vebo
+    check_aragon_doesnt_have_manage_consensus_role_on_oracle(vebo)
+    # 5) Vebo consensus version equals to 3 after voting
+    assert vebo.getConsensusVersion() == 3
+    # 7),9) Aragon agent doesnt have MANAGE_CONSENSUS_VERSION_ROLE on CS Fee oracle
+    check_aragon_doesnt_have_manage_consensus_role_on_oracle(cs_fee_oracle)
+    # 8) CS fee oracle consensus version equals to 2 after voting
+    assert cs_fee_oracle.getConsensusVersion() == 2
+    # 10) Old CS Verifier doesn't have VERIFIER_ROLE role on CSM after voting
+    assert not csm.hasRole(VERIFIER_ROLE, CS_VERIFIER_ADDRESS_OLD)
+    # 11) New CS Verifier has VERIFIER_ROLE role on CSM after voting
+    assert csm.hasRole(VERIFIER_ROLE, CS_VERIFIER_ADDRESS)
+
+    # Events check
+    display_voting_events(vote_tx)
+    events = group_voting_events(vote_tx)
+
+    assert len(events) == 11
+
+    # Validate ao consensus version set
+    validate_consensus_version_update(events[:3], AO_CONSENSUS_VERSION)
```

Comment:
Please check the executer contract for all events in this test - refer to the implementation here: https://github.com/lidofinance/scripts/blob/a9c911b78d7f40a1fb3f84568a3ccc5ecd6d08a6/utils/test/event_validators/permission.py#L89